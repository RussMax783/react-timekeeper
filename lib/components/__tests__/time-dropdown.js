'use strict';

var _enzyme = require('enzyme');

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTestRenderer = require('react-test-renderer');

var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);

var _TimeDropdown = require('../TimeDropdown');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var OPTIONS = Array.apply(null, { length: 5 }).map(function (a, i) {
	return (i + 1).toString();
});

describe('TimeDropdown component', function () {
	var closeFn = jest.fn();
	var updateValFn = jest.fn();

	test('renders correctly', function () {
		var val = 3;
		var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_TimeDropdown.TimeDropdown, {
			config: {},
			val: val,
			options: [1, 2, 3, 4, 5].map(function (v) {
				return v.toString();
			}),
			close: closeFn,
			updateVal: updateValFn
		}), {
			createNodeMock: function createNodeMock(el) {
				if (el.type === 'div') {
					// allow for setting scrollTop in `componentDidMount` method
					// otherwise throws error: `Can't add property scrollTop, object is not extensible`
					return Object.assign({}, el, { scrollTop: '' });
				}
				return el;
			}
		}).toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('triggers close and updateVal on selecting option', function () {
		var indexToSelect = 2;
		var rendered = (0, _enzyme.shallow)(_react2.default.createElement(_TimeDropdown.TimeDropdown, {
			config: {},
			val: 4,
			options: OPTIONS,
			close: closeFn,
			updateVal: updateValFn
		}));
		rendered.find('ul').childAt(indexToSelect).simulate('click');
		expect(closeFn.mock.calls.length).toBe(1);
		expect(updateValFn.mock.calls.length).toBe(1);
		expect(updateValFn).lastCalledWith((indexToSelect + 1).toString());
	});
});