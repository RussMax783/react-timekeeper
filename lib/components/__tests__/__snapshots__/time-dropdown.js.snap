exports[`TimeDropdown component renders correctly 1`] = `
<div
  className="react-timekeeper__time-dropdown"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Array [
      Object {
        "animation": "x 0.2s ease-out both",
        "animationName": Object {
          "__process": [Function],
          "__radiumKeyframes": true,
        },
        "background": "white",
        "border": "1px solid undefined",
        "borderRadius": "2px",
        "boxShadow": "0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)",
        "display": "inline-block",
        "height": "250px",
        "opacity": 0,
        "overflowY": "auto",
        "padding": "6px 0",
        "position": "absolute",
        "top": 62,
        "zIndex": "20",
      },
      undefined,
    ]
  }>
  <ul
    className="react-timekeeper__time-dropdown-options"
    style={
      Object {
        "listStyle": "none",
        "margin": 0,
        "padding": 0,
        "position": "relative",
      }
    }>
    <li
      onClick={[Function]}
      style={
        Object {
          ":hover": Object {
            "background": undefined,
          },
          "background": "",
          "color": undefined,
          "cursor": "pointer",
          "fontSize": "16px",
          "padding": "7px 30px",
        }
      }>
      1
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          ":hover": Object {
            "background": undefined,
          },
          "background": "",
          "color": undefined,
          "cursor": "pointer",
          "fontSize": "16px",
          "padding": "7px 30px",
        }
      }>
      2
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          ":hover": Object {
            "background": undefined,
          },
          "background": undefined,
          "color": undefined,
          "cursor": "pointer",
          "fontSize": "16px",
          "padding": "7px 30px",
        }
      }>
      3
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          ":hover": Object {
            "background": undefined,
          },
          "background": "",
          "color": undefined,
          "cursor": "pointer",
          "fontSize": "16px",
          "padding": "7px 30px",
        }
      }>
      4
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          ":hover": Object {
            "background": undefined,
          },
          "background": "",
          "color": undefined,
          "cursor": "pointer",
          "fontSize": "16px",
          "padding": "7px 30px",
        }
      }>
      5
    </li>
  </ul>
</div>
`;
